atecc608a_se.c

```diff
@@ -208,24 +247,68 @@ static psa_status_t atecc608a_asymmetric_sign(psa_key_slot_number_t key_slot,
     return status;
 }
 
+psa_status_t atecc608a_asymmetric_verify(psa_key_slot_number_t key_slot,
+                                         psa_algorithm_t alg,
+                                         const uint8_t *p_hash,
+                                         size_t hash_length,
+                                         const uint8_t *p_signature,
+                                         size_t signature_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool is_verified = false;
+
+    /* Keys 8 to 15 can store public keys. */
```

Comment:
“can store” — but why does this matter at this point? What matters here is what _is_ stored. Please clarify the comment or improve the code. Is this a sanity check or does it prevent failures later?

Reply:
This is a sanity check for arguments - we haven't added any validation of what exactly _is_ pre-provisioned on the chip. Should we?

Reply:
It depends primarily on what the consequences of an error are, and secondarily on how costly it is to make robust checks. At this point, how pre-provisioning works is still in an exploratory stage, so I don't have any particular requirements on how it's validated.

Here are some questions that are relevant for deciding what to do. Answering them is not a blocker for this PR, only a blocker for putting the driver into production. What else might be in slots 8–15 if people follow the driver documentation? What configuration indicates what is present in these slots and who owns these slots? Who can modify this configuration? What happens if the content of the slot does not have the right type (crash, nonsensical output, nonsensical output that leaks confidential data, …)?

Reply:
There's no driver documentation on provisioning so far. 
Initially, the slots are only limited by their size and the fact that slot 15 has an unique use counter:
- Slots 0-7 have 36 bytes (usually used for private keys);
- Slot 8 has 416 bytes (usually used for a certificate / signature / arbitrary data);
- Slots 9-14 have 72 bytes (usually used for public keys);
- Slot 15 has 72 bytes and can be a subject of a limited use counter.

Key slots may be used by the hardware only if the configuration is locked. Configuration can be read later on to determine how are the slots configured.
If an illegal operation is performed using a slot, or a slot contains bad data, an error is returned from the API.
-

Reply:
Thanks. So the check should be for slots 9–14 here. Also, please make this check an auxiliary function, since it's used in two places already.

Reply:
As this is a sanity, not a configuration check, I'd rather opt for leaving the 8-15 check here - someone can have the board configured so that the public key is in slot 8 or 15. On the contrary, it's not possible to have a key in slot 0-7.

Reply:
I've added a function, and there was a typo in the numbers anyway - fixed.